### PR TITLE
[#1975] Early work on project title stuff

### PR DESCRIPTION
**Pull request by dseif**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/e0008767d1b3930/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/e0008767d1b3930/output.txt

Total script time: 0.85 mins

Server is running on http://pr789.butyr.org/templates/basic/
Run the Test Suite: http://pr789.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/4dfa653510a3e21/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/4dfa653510a3e21/output.txt

Total script time: 1.13 mins

Server is running on http://pr789.butyr.org/templates/basic/
Run the Test Suite: http://pr789.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/8359bab6f793971/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/8359bab6f793971/output.txt

Total script time: 0.65 mins

Server is running on http://pr789.butyr.org/templates/basic/
Run the Test Suite: http://pr789.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/46157d550840485/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @mjschranz received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/f436add9387ffcc/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/46157d550840485/output.txt

Total script time: 0.86 mins

Server is running on http://pr789.butyr.org/templates/basic/
Run the Test Suite: http://pr789.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/f436add9387ffcc/output.txt

Total script time: 1.14 mins

File: src/ui/header.js [hunk 1]
```diff
@@ -115,10 +123,69 @@ define( [ "dialog/dialog", "util/lang", "text!layouts/header.html" ],
...
+      _noProjectNameToolTip.classList.add( "tooltip-error" );
+      _projectName.addEventListener( "mouseover", onMouseOver, false );
+      return true;
+    }
+
+    function onBlur() {
+      var node = _projectName.childNodes[ 1 ],
+          text = document.createTextNode( node.value || _projectNamePlaceHolderText );
+
+      butter.project.name = node.value;
```
Comment (humphd) [inline src/ui/header.js]: Should we attempt to do any sanitization or cleaning of this?

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @humphd received. Current queue size: 2

Live output at: http://botio.butyr.org:8000/041c762db69699e/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/041c762db69699e/output.txt

Total script time: 0.60 mins

Server failed to start

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @humphd received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/7af3a8f3c980e09/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/7af3a8f3c980e09/output.txt

Total script time: 0.53 mins

Server failed to start

**Comment by jbuck** [comment]
Staged on master: https://github.com/mozilla/butter/commit/21da61657dad419f0a44db7e3a4ce60e69f1eb85

**Pull request closed without merging**